src/bundle/ui-dev/src/modules/common/popup/popup.component.js

```diff
@@ -16,205 +18,128 @@ const MODAL_SIZE_CLASS = {
     large: 'modal-lg',
 };
 
-class Popup extends Component {
-    constructor(props) {
-        super(props);
-
-        this._refModal = null;
-
-        this.setModalRef = this.setModalRef.bind(this);
-        this.onKeyUp = this.onKeyUp.bind(this);
-
-        this.state = {
-            currentProps: {
-                isVisible: props.isVisible,
-                isLoading: props.isLoading,
-            },
-            isVisible: props.isVisible,
-            isLoading: props.isLoading,
-        };
-    }
-
-    static getDerivedStateFromProps(nextProps, prevState) {
-        if (nextProps.isVisible !== prevState.currentProps.isVisible || nextProps.isLoading !== prevState.currentProps.isLoading) {
-            return {
-                currentProps: {
-                    isVisible: nextProps.isVisible,
-                    isLoading: nextProps.isLoading,
-                },
-                isVisible: nextProps.isVisible,
-                isLoading: nextProps.isLoading,
-            };
-        }
-
+const Popup = ({
+    isVisible,
+    onClose,
+    children,
+    title,
+    subtitle,
+    hasFocus,
+    noKeyboard,
+    actionBtnsConfig,
+    size,
+    noHeader,
+    noCloseBtn,
+    extraClasses,
+}) => {
+    if (!isVisible) {
         return null;
     }
 
-    componentDidMount() {
-        const { noKeyboard, hasFocus } = this.props;
-        const { isVisible: show } = this.state;
-
-        if (show) {
-            const bootstrapModal = window.bootstrap.Modal.getOrCreateInstance(this._refModal, {
-                ...MODAL_CONFIG,
-                keyboard: !noKeyboard,
-                focus: hasFocus,
-            });
-
-            bootstrapModal.show();
-
-            this.attachModalEventHandlers();
-        }
-    }
-
-    componentDidUpdate() {
-        const { isVisible: show } = this.state;
-
-        const bootstrapModal = window.bootstrap.Modal.getOrCreateInstance(this._refModal, {
+    const modalRef = useRef(null);
```

Comment:
You have sonar bug here - as all hooks should be declared before any if (and not in loops, but that's not case here), in theory current way won't break anything as you have return if at the beginning.

But to appease sonar I would move this one line before if (line 35)

Reply:
Ye but there is still useEffect :/